packages/features/liquidity/src/liquidity-variable/hooks/use-liquidity-pools-table.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+  DataTableColumn,
+  MultiValueCell,
+  MultiValueIconCell,
+  TxnHashCell,
+} from '@notional-finance/mui';
+import { useNotionalContext } from '@notional-finance/notionable-hooks';
+import { useContext } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { LiquidityContext } from '../../liquidity';
+import {
+  RATE_PRECISION,
+  formatMaturity,
+  getEtherscanTransactionLink,
+} from '@notional-finance/util';
+import { formatNumberAsPercent } from '@notional-finance/helpers';
+import moment from 'moment';
+import {
+  Registry,
+  TokenBalance,
+  fCashMarket,
+} from '@notional-finance/core-entities';
+
+export const useLiquidityPoolsTable = () => {
+  const {
+    state: { deposit, selectedNetwork },
+  } = useContext(LiquidityContext);
+  const {
+    globalState: { historicalTrading },
+  } = useNotionalContext();
+  let poolTableData: Record<string, any>[] = [];
+
+  if (deposit && deposit?.currencyId && selectedNetwork && historicalTrading) {
+    poolTableData = historicalTrading[selectedNetwork][deposit?.currencyId]
```

Comment:
Can you just do all this data processing in the `on-data-update` method? That way we will only re-calculate all this data periodically rather than on every render.